src/ldp/http/metadata/LinkTypeParser.ts

```diff
@@ -0,0 +1,38 @@
+import { DataFactory } from 'n3';
+import type { HttpRequest } from '../../../server/HttpRequest';
+import { UnsupportedHttpError } from '../../../util/errors/UnsupportedHttpError';
+import { parseParameters, splitAndClean, transformQuotedStrings } from '../../../util/HeaderUtil';
+import { RDF } from '../../../util/UriConstants';
+import type { RepresentationMetadata } from '../../representation/RepresentationMetadata';
+import type { MetadataParser } from './MetadataParser';
+
+/**
+ * Parses Link headers with "rel=type" parameters and adds them as RDF.type metadata.
+ */
+export class LinkTypeParser implements MetadataParser {
+  public async parse(request: HttpRequest, metadata: RepresentationMetadata): Promise<void> {
+    const { link } = request.headers;
+    if (link) {
+      if (Array.isArray(link)) {
+        link.forEach((entry): void => this.parseLink(entry, metadata));
+      } else {
+        this.parseLink(link, metadata);
+      }
+    }
+  }
+
+  private parseLink(linkEntry: string, metadata: RepresentationMetadata): void {
+    const { result, replacements } = transformQuotedStrings(linkEntry);
+    splitAndClean(result).forEach((part): void => {
```

Comment:
So I typically recommend

```suggestion
    for (const part of splitAndClean(result)) {
```

Since the introduction of local scoping with `const` and `let`, there are no scoping reasons anymore to prefer functions over loops, and the latter would never be slower (but can be faster).